src/main/java/org/homer/versioner/core/procedure/RelationshipProcedure.java

```diff
@@ -28,6 +30,36 @@
  */
 public class RelationshipProcedure extends CoreProcedure {
 
+    @Procedure(value = "graph.versioner.relationships.create", mode = Mode.WRITE)
+    @Description("graph.versioner.relationships.create(entityA, entitiesB, relProps, date) - Create multiple relationships from entitySource to each of the entityDestinations with the given type and/or properties for the specified date.  The relationship 'versionerLabel' along with properties for each relationship can be passed in via 'relProps' a default label ('LABEL_UNDEFINED') is assigned to relationships that are not supplied with a 'versionerLabel' attribute in the props")
+    public Stream<RelationshipOutput> relationshipsCreate(
+            @Name("entitySource") Node entitySource,
+            @Name("entityDestinations") List<Node> entityDestinations,
+            @Name(value = "relProps", defaultValue = "[{}]") List<Map<String, Object>> relProps,
+            @Name(value = "date", defaultValue = "null") LocalDateTime date) {
+
+        Optional<Node> sourceCurrentState = createNewSourceState(entitySource, defaultToNow(date));
+        isEntityOrThrowException(entitySource);
+
+        return entityDestinations.size() == relProps.size() ? sourceCurrentState.map(node -> zip(entityDestinations, relProps).stream().map((item) -> {
```

Comment:
Is it possible to use explicit "if" condition? It is more readable.

Reply:
please see latest commit